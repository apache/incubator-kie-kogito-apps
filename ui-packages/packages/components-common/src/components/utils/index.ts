export * from './FormActionsUtils';
export * from './FormValidator';
export * from './ModelConversionTool';
