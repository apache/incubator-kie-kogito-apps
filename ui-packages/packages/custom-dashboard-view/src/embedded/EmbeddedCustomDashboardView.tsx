import React, { useCallback } from 'react';
import { EnvelopeServer } from '@kie-tools-core/envelope-bus/dist/channel';
import {
  EmbeddedEnvelopeProps,
  RefForwardingEmbeddedEnvelope
} from '@kie-tools-core/envelope/dist/embedded';
import {
  CustomDashboardViewApi,
  CustomDashboardViewChannelApi,
  CustomDashboardViewEnvelopeApi,
  CustomDashboardViewDriver
} from '../api';
import { CustomDashboardViewChannelApiImpl } from './CustomDashboardViewChannelApiImpl';
import { ContainerType } from '@kie-tools-core/envelope/dist/api';
import { init } from '../envelope';

export interface Props {
  targetOrigin: string;
  driver: CustomDashboardViewDriver;
  dashboardName: string;
}

export const EmbeddedCustomDashboardView = React.forwardRef(
  (props: Props, forwardedRef: React.Ref<CustomDashboardViewApi>) => {
    const refDelegate = useCallback(
      (
        envelopeServer: EnvelopeServer<
          CustomDashboardViewChannelApi,
          CustomDashboardViewEnvelopeApi
        >
      ): CustomDashboardViewApi => ({}),
      []
    );
    const pollInit = useCallback(
      (
        envelopeServer: EnvelopeServer<
          CustomDashboardViewChannelApi,
          CustomDashboardViewEnvelopeApi
        >,
        container: () => HTMLDivElement
      ) => {
        init({
          config: {
            containerType: ContainerType.DIV,
            envelopeId: envelopeServer.id
          },
          container: container(),
          bus: {
            postMessage(message, targetOrigin, transfer) {
              window.postMessage(message, targetOrigin, transfer);
            }
          }
        });
        return envelopeServer.envelopeApi.requests.customDashboardView__init(
          {
            origin: envelopeServer.origin,
            envelopeServerId: envelopeServer.id
          },
          props.dashboardName
        );
      },
      []
    );

    return (
      <EmbeddedCustomDashboardViewEnvelope
        ref={forwardedRef}
        apiImpl={new CustomDashboardViewChannelApiImpl(props.driver)}
        origin={props.targetOrigin}
        refDelegate={refDelegate}
        pollInit={pollInit}
        config={{ containerType: ContainerType.DIV }}
      />
    );
  }
);

const EmbeddedCustomDashboardViewEnvelope = React.forwardRef<
  CustomDashboardViewApi,
  EmbeddedEnvelopeProps<
    CustomDashboardViewChannelApi,
    CustomDashboardViewEnvelopeApi,
    CustomDashboardViewApi
  >
>(RefForwardingEmbeddedEnvelope);
