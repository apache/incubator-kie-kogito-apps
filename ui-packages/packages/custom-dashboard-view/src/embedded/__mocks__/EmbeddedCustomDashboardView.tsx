import React from 'react';

const MockedEmbeddedCustomDashboardView = (): React.ReactElement => {
  return <></>;
};

export default MockedEmbeddedCustomDashboardView;
