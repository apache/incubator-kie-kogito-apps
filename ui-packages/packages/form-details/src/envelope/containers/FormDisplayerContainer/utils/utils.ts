import { Form } from '../../../../api';

import uuidv4 from 'uuid';

export function buildTestContext(form: Form) {
  return {
    user: {
      id: 'test',
      groups: ['group1', 'group2']
    },
    schema: JSON.parse(form.configuration.schema),
    task: {
      id: uuidv4(),
      description: 'task description',
      name: 'TaskName',
      referenceName: 'Task Name',
      priority: '1',
      processInstanceId: uuidv4(),
      processId: 'process',
      rootProcessInstanceId: null,
      rootProcessId: null,
      state: 'Ready',
      completed: null,
      started: Date.now(),
      lastUpdate: Date.now()
    },
    phases: ['abort', 'claim', 'complete', 'skip']
  };
}
