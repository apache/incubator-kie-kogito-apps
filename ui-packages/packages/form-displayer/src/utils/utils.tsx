/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const renderTags = (container, resources): void => {
  for (const key in resources.scripts) {
    const script: HTMLScriptElement = document.createElement('script');

    script.src = resources.scripts[key];
    container.appendChild(script);
  }

  for (const key in resources.styles) {
    const link: HTMLLinkElement = document.createElement('link');
    link.rel = 'stylesheet';
    link.href = resources.styles[key];
    container.appendChild(link);
  }
};

export const renderResources = (containerId, resources): void => {
  const container: HTMLElement = document.getElementById(containerId);
  const scripts: HTMLCollectionOf<HTMLScriptElement> = container.getElementsByTagName(
    'script'
  );
  const styles: HTMLCollectionOf<HTMLLinkElement> = container.getElementsByTagName(
    'link'
  );
  if (scripts.length > 0 || styles.length > 0) {
    let scriptIndex: number = scripts.length;
    let styleIndex: number = styles.length;
    while (scriptIndex--) {
      container.removeChild(scripts[scriptIndex]);
    }
    while (styleIndex--) {
      container.removeChild(styles[styleIndex]);
    }
    renderTags(container, resources);
  } else {
    renderTags(container, resources);
  }
};

export const sourceHandler = source => {
  const reactReg = /import React, {[^}]*}.*(?='react').*/gim;
  const patternflyReg = /import {[^}]*}.*(?='@patternfly\/react-core').*/gim;
  const regexvalueReact = new RegExp(reactReg);
  const reactImport = regexvalueReact.exec(source);
  const reg = /\{([^)]+)\}/;
  const reactElements = reg.exec(reactImport[0])[1];
  const regexvaluePat = new RegExp(patternflyReg);
  const patternflyImport = regexvaluePat.exec(source);
  const patternflyElements = reg.exec(patternflyImport[0])[1];
  const trimmedSource = source
    .split(reactReg)
    .join('')
    .trim()
    .split(patternflyReg)
    .join('')
    .trim();
  const tempSource = trimmedSource;
  const formName = tempSource.split(':')[0].split('const ')[1];

  return { reactElements, patternflyElements, formName, trimmedSource };
};
