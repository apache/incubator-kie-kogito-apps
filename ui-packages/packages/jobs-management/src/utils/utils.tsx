import React from 'react';
import { JobStatus } from '@kogito-apps/management-console-shared/dist/types';
import { UndoIcon } from '@patternfly/react-icons/dist/js/icons/undo-icon';
import { ClockIcon } from '@patternfly/react-icons/dist/js/icons/clock-icon';
import { ErrorCircleOIcon } from '@patternfly/react-icons/dist/js/icons/error-circle-o-icon';
import { BanIcon } from '@patternfly/react-icons/dist/js/icons/ban-icon';
import { CheckCircleIcon } from '@patternfly/react-icons/dist/js/icons/check-circle-icon';

export const JobsIconCreator = (state: JobStatus): JSX.Element => {
  switch (state) {
    case JobStatus.Error:
      return (
        <>
          <ErrorCircleOIcon
            className="pf-u-mr-sm"
            color="var(--pf-global--danger-color--100)"
          />
          Error
        </>
      );
    case JobStatus.Canceled:
      return (
        <>
          <BanIcon className="pf-u-mr-sm" />
          Canceled
        </>
      );
    case JobStatus.Executed:
      return (
        <>
          <CheckCircleIcon
            className="pf-u-mr-sm"
            color="var(--pf-global--success-color--100)"
          />
          Executed
        </>
      );
    case JobStatus.Retry:
      return (
        <>
          <UndoIcon className="pf-u-mr-sm" />
          Retry
        </>
      );
    case JobStatus.Scheduled:
      return (
        <>
          <ClockIcon className="pf-u-mr-sm" />
          Scheduled
        </>
      );
  }
};
