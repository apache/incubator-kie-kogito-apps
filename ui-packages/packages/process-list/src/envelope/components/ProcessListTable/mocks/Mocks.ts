/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import {
  ProcessInstance,
  ProcessInstanceState
} from '@kogito-apps/management-console-shared';
export const ProcessInstances: ProcessInstance[] = [
  {
    id: '538f9feb-5a14-4096-b791-2055b38da7c6',
    processId: 'travels',
    businessKey: 'Tra234',
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    rootProcessInstanceId: null,
    roles: [],
    state: ProcessInstanceState.Error,
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    start: new Date('2019-10-22T03:40:44.089Z'),
    error: {
      nodeDefinitionId: '__a1e139d5-4e77-48c9-84ae-34578e9817n',
      message: 'Something went wrong'
    },
    lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-23T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-23T22:00:00Z[UTC]","city":"New York","country":"US","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"New York","country":"US","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Berlin","country":"Germany","street":"Bakers","zipCode":"100200"},"email":"cristiano@redhat.com","firstName":"Cristiano","lastName":"Nicolai","nationality":"German"}}',
    nodes: [],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21eajabbcc',
        processId: 'travels',
        businessKey: 'TP444',
        parentProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        parentProcessInstance: null,
        rootProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        processName: 'FlightBooking test 2',
        roles: [],
        state: ProcessInstanceState.Active,
        serviceUrl: 'http://localhost:4000',
        lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
        endpoint: 'http://localhost:4000',
        addons: ['process-management'],
        error: {
          nodeDefinitionId: 'a1e139d5-81c77-48c9-84ae-34578e90433n',
          message: 'Something went wrong'
        },
        start: new Date('2019-10-22T03:40:44.089Z'),
        end: new Date('2019-10-22T03:40:44.089Z'),
        variables:
          '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
        nodes: [],
        milestones: [],
        childProcessInstances: []
      }
    ]
  },
  {
    id: '538f9feb-5a14-4096-b791-2055b38da7c6',
    processId: 'travels',
    businessKey: 'Tra234',
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    rootProcessInstanceId: null,
    roles: [],
    state: ProcessInstanceState.Error,
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    start: new Date('2019-10-22T03:40:44.089Z'),
    error: {
      nodeDefinitionId: '__a1e139d5-4e77-48c9-84ae-34578e9817n',
      message: 'Something went wrong'
    },
    lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-23T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-23T22:00:00Z[UTC]","city":"New York","country":"US","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"New York","country":"US","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Berlin","country":"Germany","street":"Bakers","zipCode":"100200"},"email":"cristiano@redhat.com","firstName":"Cristiano","lastName":"Nicolai","nationality":"German"}}',
    nodes: [],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21eajabbcc',
        processId: 'travels',
        businessKey: 'TP444',
        parentProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        parentProcessInstance: null,
        rootProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        processName: 'FlightBooking test 2',
        roles: [],
        state: ProcessInstanceState.Active,
        serviceUrl: 'http://localhost:4000',
        lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
        endpoint: 'http://localhost:4000',
        addons: ['process-management'],
        error: {
          nodeDefinitionId: 'a1e139d5-81c77-48c9-84ae-34578e90433n',
          message: 'Something went wrong'
        },
        start: new Date('2019-10-22T03:40:44.089Z'),
        end: new Date('2019-10-22T03:40:44.089Z'),
        variables:
          '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
        nodes: [],
        milestones: [],
        childProcessInstances: []
      }
    ]
  }
];
