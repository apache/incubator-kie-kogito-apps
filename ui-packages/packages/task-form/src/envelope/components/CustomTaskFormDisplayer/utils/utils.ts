import cloneDeep from 'lodash/cloneDeep';
import unset from 'lodash/unset';
import { User } from '../../../../api';
import { UserTaskInstance } from '@kogito-apps/task-console-shared';

export function buildTaskFormContext(
  userTask: UserTaskInstance,
  schema: Record<string, any>,
  user: User
): Record<string, any> {
  const ctxSchema = cloneDeep(schema);

  const ctxPhases = ctxSchema.phases;

  unset(ctxSchema, 'phases');

  const ctxTask = cloneDeep(userTask);

  unset(ctxTask, 'actualOwner');
  unset(ctxTask, 'adminGroups');
  unset(ctxTask, 'adminUsers');
  unset(ctxTask, 'excludedUsers');
  unset(ctxTask, 'potentialGroups');
  unset(ctxTask, 'potentialUsers');
  unset(ctxTask, 'inputs');
  unset(ctxTask, 'outputs');
  unset(ctxTask, 'endpoint');

  return {
    user: user,
    task: ctxTask,
    schema: ctxSchema,
    phases: ctxPhases
  };
}
