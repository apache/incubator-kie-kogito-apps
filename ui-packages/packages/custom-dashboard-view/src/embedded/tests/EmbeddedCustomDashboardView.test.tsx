import React from 'react';
import { EmbeddedCustomDashboardView } from '../EmbeddedCustomDashboardView';
import { MockedCustomDashboardViewDriver } from './utils/Mocks';
import { render } from '@testing-library/react';

describe('EmbeddedCustomDashboardView tests', () => {
  it('Snapshot', () => {
    const props = {
      dashboardName: 'name',
      targetOrigin: 'origin',
      driver: new MockedCustomDashboardViewDriver()
    };

    const { container } = render(<EmbeddedCustomDashboardView {...props} />);

    const contentDiv = container.querySelector('div');

    expect(contentDiv).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
