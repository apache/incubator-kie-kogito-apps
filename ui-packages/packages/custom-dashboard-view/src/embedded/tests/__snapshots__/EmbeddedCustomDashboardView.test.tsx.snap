// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbeddedCustomDashboardView tests Snapshot 1`] = `
<div>
  <div />
</div>
`;
